### PR TITLE
added route change announcement

### PR DESCRIPTION
**Pull request by JanAckermann**
## Description
Enhancement: Implement live region updates on route changes

## Related Issue

- Fixes https://github.com/owncloud/web/issues/4346

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by JanAckermann** [comment]
look at div announcer :rocket:

**Comment by JanAckermann** [comment]
@pascalwengerter @kulmann Could you have a closer look?
* I removed vue-announcer and replaced it with the oc-hidden-announcer
* Needed to add i18n to the web-runtime router

**Comment by JanAckermann** [comment]
Dunno whats up with license/cla

**Comment by JanAckermann** [comment]
@kulmann i18n interpolate introduced

File: packages/web-runtime/src/App.vue [hunk 1]
```diff
@@ -321,11 +324,19 @@ export default {
...
     },
-    extractPageTitleFromRoute(route) {
-      const titleSegments = [this.configuration.theme.general.name]
 
-      if (route.meta.pageTitle) {
-        titleSegments.unshift(route.meta.pageTitle)
+    announceRouteChange(route) {
+      const pageTitle = this.extractPageTitleFromRoute(route, false)
+
+      const translated = this.$gettext(' Navigated to %{ pageTitle }')
```
Comment (kulmann) [inline packages/web-runtime/src/App.vue:331]: Why the blank before `Navigate`?

File: packages/web-runtime/src/App.vue [hunk 2]
```diff
@@ -83,7 +84,7 @@
 </template>
 <script>
 import 'inert-polyfill'
-import { mapGetters, mapState, mapActions } from 'vuex'
+import {mapActions, mapGetters, mapState} from 'vuex'
```
Comment (kulmann) [inline packages/web-runtime/src/App.vue:87]:
```suggestion
import { mapActions, mapGetters, mapState } from 'vuex'
```
... to make the linter happy ;-)

File: packages/web-runtime/src/App.vue [hunk 3]
```diff
@@ -321,11 +324,18 @@ export default {
...
       this.appNavigationVisible = false
     },
-    extractPageTitleFromRoute(route) {
-      const titleSegments = [this.configuration.theme.general.name]
 
-      if (route.meta.pageTitle) {
-        titleSegments.unshift(route.meta.pageTitle)
+    announceRouteChange(route) {
+      const pageTitle = this.extractPageTitleFromRoute(route, false)
+      const translated = this.$gettext(' Navigated to %{ pageTitle }')
```
Comment (kulmann) [inline packages/web-runtime/src/App.vue:330]:
```suggestion
      const translated = this.$gettext('Navigated to %{ pageTitle }')
```
sorry, I meant this blank ;-)

**Review by kulmann — APPROVED** [review]
💪

**Comment by kulmann** [comment]
Added @marcus-herrmann as reviewer, since I lack experience with the live region updates and cannot tell if this is sufficient. Code looks good. Could you take a look @marcus-herrmann ?

**Review by marcus-herrmann — APPROVED** [review]
LGTM. But is it possible to see this working live soon?

**Comment by kulmann** [comment]
> LGTM. But is it possible to see this working live soon?

As soon as it's merged it will be grabbed by the next auto-reset on web.owncloud.com. Happens at least once per day afaik. So if we can merge it today it would be visible by tomorrow.

**Comment by marcus-herrmann** [comment]
Ok. So web.owncloud.com is newer than ocis-a11y.owncloud.com then. Interestingly I see a different side wide search function there (compared to ocis-a11y where I audited). Also, the files table fix doesn't seem to be on "web."

**Comment by kulmann** [comment]
> Ok. So web.owncloud.com is newer than ocis-a11y.owncloud.com then. Interestingly I see a different side wide search function there (compared to ocis-a11y where I audited). Also, the files table fix doesn't seem to be on "web."

ocis doesn't have search, that's why it's disabled there. Files table is still not merged. ;-) The PR is close to being merged though.

**Comment by marcus-herrmann** [comment]
> ocis doesn't have search, that's why it's disabled there. Files table is still not merged. ;-) The PR is close to being merged though.

Oh I see. But if ocis-a11y is not representative (at least in the regards of search), it is not ideal as a basis for an accessibility audit?

**Comment by marcus-herrmann** [comment]
> Oh I see. But if ocis-a11y is not representative (at least in the regards of search), it is not ideal as a basis for an accessibility audit?

Because I see a WCAG 3.3.2 violation there: https://www.w3.org/TR/WCAG20-TECHS/H32.html (sorry for hijacking the ticket)

**Review by pascalwengerter — APPROVED** [review]
LGTM! 🔥

**Comment by pascalwengerter** [comment]
Works well for me, locally. @marcus-hermann can this be merged and is your concerns regarding a possible `WCAG 3.3.2 violation` (I guess regarding the search submit, but not sure if related to route change announcements?) tracked somewhere already?

**Comment by kulmann** [comment]
Merging, as the comment about search was not related to the announcer. Question about whether or not it is tracked somewhere is still valid though. 😅

**Comment by marcus-herrmann** [comment]
> Works well for me, locally. @marcus-hermann can this be merged and is your concerns regarding a possible WCAG 3.3.2 violation (I guess regarding the search submit, but not sure if related to route change announcements?) tracked somewhere already?

@pascalwengerter: Sorry, this is unrelated to this ticket, so it would be absurd  to have been a merge blocker. I will create a new issue, though, since this could/will be critical for WCAG/BITV but was not part of the version I audited.

**Comment by pascalwengerter** [comment]
> > Works well for me, locally. @marcus-hermann can this be merged and is your concerns regarding a possible WCAG 3.3.2 violation (I guess regarding the search submit, but not sure if related to route change announcements?) tracked somewhere already?
> 
> @pascalwengerter: Sorry, this is unrelated to this ticket, so it would be absurd to have been a merge blocker. I will create a new issue, though, since this could/will be critical for WCAG/BITV but was not part of the version I audited.

Was more of a rethoric question ;) thank you & looking forward to the new ticket (and more information on what's to be fixed)!

**Comment by marcus-herrmann** [comment]
> Was more of a rethoric question ;) thank you & looking forward to the new ticket (and more information on what's to be fixed)!

Regarding that: https://github.com/owncloud/web/issues/4867

**Pull request merged**